### ISSUE TITLE
Correctly implement `ManagedNumberField` behavior (described)

### ISSUE THREAD

**Issue by knod** [codeforboston/cliff-effects#630]
This is what I've got so far for `<ManagedNumberField>` desired behavior.

**Valid number:** A positive decimal or whole number, 0, or a blank field.

- User clicks on field with '0' in it:
  - Field goes blank, ready for input
- User clicks on field with valid number in it:
  - Nothing changes
- User types a valid number in the field:
  - Typing shows up as entered
- User types a valid number in the field and then leaves the field:
  - Valid value is shown as decimal value
  - (Valid value is stored permanently)
- User types an INVALID number in the field:
  - Typing shows up as entered
  - Field gives 'invalid value' feedback (turns red)
- User types in an INVALID number and then leaves the field
  - The last valid field value previously entered is shown (last valid value typed, or last valid value left there?)
  - (Invalid value is discarded)
- User types 0 in a field:
  - Typing shows up as entered
- User deletes all items in the field
  - Input does not turn red
- User deletes all numbers in the field then leaves the field
  - Field shows '0'
  - (0 is stored permanently as the value)

Is this behavior possible?

**Comment by knod** [comment]
It looks like that behavior isn't possible with the system we currently have. Both an empty field and invalid values like 'e' return an empty string. New suggestion:

Just completely prevent the user from typing in anything other than a number with, at most, one decimal place. This includes allowing 0. It's not what we originally wanted, but maybe it's something that can actually work.

**Comment by knod** [comment]
What I found from experiments:

```
Click 0, click out (no feedback)
focus:      input value: 0.00,      value: 0
blur:       input value: ,          value: 0

Click 0, 2, click out (no feedback)
focus:      input value: 0.00,      value: 0
change:     input value: 2,         value: 0
blur:       input value: 2,         value: 2

Click 2, add .5, click out (no feedback)
focus:      input value: 2.00,      value: 2
change:     input value: 2.50,      value: 2
blur:       input value: 2.50,      value: 2.5

Click 2.5, add e, click out ('invalid' feedback)
focus:      input value: 2.50,      value: 2.5
change:     input value: ,          value: 2.5
blur:       input value: ,          value: 2.5

Click 2, delete all, click out ('invalid' feedback)
focus:      input value: 2.00,      value: 2
change:     input value: ,          value: 2
blur:       input value: ,          value: 2

Click 0, 0, click out (no feedback)
focus:      input value: 0.00,      value: 0
change:     input value: 0,         value: 0
blur:       input value: 0,         value: 0

Click 0, add e, click out (no feedback) (why no feedback?)
focus:      input value: 0.00,      value: 0
(no change event)
blur:       input value: ,          value: 0
```

I couldn't find any other differences in the event data or input props data that gave more information to differentiate something with 'e' in it and an empty field.

**Comment by knod** [comment]
@shp54 , @snyderc , @wacii , @ethanbb , @misspran: Please react to this if you would be able to take this on this week. Also feel free to suggest others who may be interested.

**Comment by knod** [comment]
We may have to make this a text field instead of a number field to be able to effectively intercept the right/wrong inputs.

**Comment by knod** [comment]
Thanks for responding! I've made an executive decision and given this one to wacii.

**Comment by knod** [comment]
@ethanbb : I think it almost does! I apologize. I'm pretty scattered right now, so things may go sideways sometimes. We could try to tell me to just stop trying to butt in for a while, but I've never had much luck following instructions like that. I'll try to be more careful.

**Comment by ethanbb** [comment]
OK no big deal! I thought maybe you just hadn't seen it yet.

**Comment by knod** [comment]
Fixed by number #629, I believe. Now we just need tests for this behavior!
